src/pages/people-page/people.page.tsx

```diff
@@ -0,0 +1,123 @@
+      });
+  }, []);
+
+  const handlePersonClick = (slug: string | null) => {
+    setSelectedPersonSlug(slug);
+  };
+
+  const shouldApplyBackground = (slug: string) =>
+    targetPersonSlug === slug ? 'has-background-warning' : '';
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+      <div className="block">
+        <div className="box table-container">
+          {isLoading ? (
+            <Loader />
+          ) : error ? (
+            <p data-cy="peopleLoadingError" className="has-text-danger">
+              Something went wrong
+            </p>
+          ) : people.length === 0 ? (
+            <p data-cy="noPeopleMessage">There are no people on the server</p>
+          ) : (
+            <table
+              data-cy="peopleTable"
+              className="table is-striped is-hoverable is-narrow is-fullwidth"
+            >
+              <thead>
+                <tr>
+                  <th>Name</th>
+                  <th>Sex</th>
+                  <th>Born</th>
+                  <th>Died</th>
+                  <th>Mother</th>
+                  <th>Father</th>
+                </tr>
+              </thead>
+              <tbody>
+                {people.map(person => (
+                  <tr
+                    key={person.slug}
+                    data-cy="person"
+                    className={shouldApplyBackground(person.slug)}
+                  >
+                    <td className={shouldApplyBackground(person.slug)}>
+                      <PersonLink
+                        person={person}
+                        onClick={() => handlePersonClick(person.slug)}
+                      >
+                        {person.name}
+                      </PersonLink>
+                    </td>
+                    <td className={shouldApplyBackground(person.slug)}>
+                      {person.sex}
+                    </td>
+                    <td className={shouldApplyBackground(person.slug)}>
+                      {person.born}
+                    </td>
+                    <td className={shouldApplyBackground(person.slug)}>
+                      {person.died}
+                    </td>
+                    <td className={shouldApplyBackground(person.slug)}>
+                      <PersonLink
+                        person={people.find(p => p.name === person.motherName)}
+                        onClick={() => handlePersonClick(person.motherName)}
+                      >
+                        {person.motherName ? person.motherName : '-'}
+                      </PersonLink>
+                    </td>
+                    <td
+                      className={
+                        selectedPersonSlug === person.slug
+                          ? 'has-background-warning'
+                          : ''
+                      }
+                    >
+                      <PersonLink
+                        person={people.find(p => p.name === person.fatherName)}
```

Comment:
This works just fine, but there is another approach that might be better in performance:
Instead of using _find_ method in all PersonLinks, which iterates the whole array until it finds a result, you could use a single iteration using a _.reduce_ method and creating either a _Map_ or an _Object_ to access it by keys, e.g.:
```
const peopleMap = people.reduce((map, person) => ({ ...map, [person.name]: person }), {});
...
<PersonLink person={peopleMap[p.fatherName]}
```
This way, you reduce N iterations in _people_ array into a single iteration (reduce method iteration) and then access the parents by using their hash, which is much more performatic than using a _.find_